test/listShards.js

```diff
@@ -0,0 +1,120 @@
+        'Member must have value greater than or equal to 1',
+        'Value \'\' at \'exclusiveStartShardId\' failed to satisfy constraint: ' +
+        'Member must satisfy regular expression pattern: [a-zA-Z0-9_.-]+',
+        'Value \'\' at \'exclusiveStartShardId\' failed to satisfy constraint: ' +
+        'Member must have length greater than or equal to 1',
+        'Value \'\' at \'streamName\' failed to satisfy constraint: ' +
+        'Member must satisfy regular expression pattern: [a-zA-Z0-9_.-]+',
+        'Value \'\' at \'streamName\' failed to satisfy constraint: ' +
+        'Member must have length greater than or equal to 1',
+      ], done)
+    })
+
+    it('should return ValidationException for long StreamName', function(done) {
+      var name = new Array(129 + 1).join('a')
+      assertValidation({StreamName: name, Limit: 100000, ExclusiveStartShardId: name}, [
+        'Value \'100000\' at \'limit\' failed to satisfy constraint: ' +
+        'Member must have value less than or equal to 10000',
+        'Value \'' + name + '\' at \'exclusiveStartShardId\' failed to satisfy constraint: ' +
+        'Member must have length less than or equal to 128',
+        'Value \'' + name + '\' at \'streamName\' failed to satisfy constraint: ' +
+        'Member must have length less than or equal to 128',
+      ], done)
+    })
+
+    it('should return ResourceNotFoundException if stream does not exist', function(done) {
+      var name = randomName()
+      assertNotFound({StreamName: name}, 'Stream ' + name + ' under account ' +
+        helpers.awsAccountId + ' not found.', done)
+    })
+
+  })
+
+  describe('functionality', function() {
+
+    it('should return list of shards', function(done) {
+      request(opts({StreamName: helpers.testStream}), function(err, res) {
+        if (err) return done(err)
+        res.statusCode.should.equal(200)
+
+        res.body.Shards[0].SequenceNumberRange.StartingSequenceNumber.should.match(/^\d{56}$/)
+        res.body.Shards[1].SequenceNumberRange.StartingSequenceNumber.should.match(/^\d{56}$/)
+        res.body.Shards[2].SequenceNumberRange.StartingSequenceNumber.should.match(/^\d{56}$/)
+
+        delete res.body.Shards[0].SequenceNumberRange.StartingSequenceNumber
+        delete res.body.Shards[1].SequenceNumberRange.StartingSequenceNumber
+        delete res.body.Shards[2].SequenceNumberRange.StartingSequenceNumber
+
+        res.body.should.eql({
+            Shards: [{
+              ShardId: 'shardId-000000000000',
+              SequenceNumberRange: {},
+              HashKeyRange: {
+                StartingHashKey: '0',
+                EndingHashKey: '113427455640312821154458202477256070484',
+              },
+            }, {
+              ShardId: 'shardId-000000000001',
+              SequenceNumberRange: {},
+              HashKeyRange: {
+                StartingHashKey: '113427455640312821154458202477256070485',
```

Comment:
Are these values mocked somewhere? Will they change ever?

Reply:
The number of shards is stored in ./helper.js. The hash key is an implementation details and is guaranteed to be those values. (Hash values is typically @^64 or 2^128 / number of shards).